src/modules/user/user.usecase.ts

```diff
@@ -518,4 +558,36 @@ export class UserUseCases {
       password,
     });
   }
+
+  async updatePassword(
+    user: User,
+    updatePasswordDto: UpdatePasswordDto,
+  ): Promise<void> {
+    const { currentPassword, newPassword, newSalt, mnemonic, privateKey } =
+      updatePasswordDto;
+
+    if (user.password.toString() !== currentPassword) {
+      throw new InvalidPasswordError();
+    }
+    await this.userRepository.updateById(user.id, {
+      password: newPassword,
+      hKey: new Buffer(newSalt),
+      mnemonic,
+    });
+
+    const [keyServer] = await this.keyServerRepository.findUserKeysOrCreate(
+      user.id,
+      {
+        encryptVersion: privateKey,
+      },
+    );
+
+    if (!keyServer) {
+      throw new KeyServerNotFoundError();
+    }
+
+    await this.keyServerRepository.update(user.id, {
+      encryptVersion: privateKey,
```

Comment:
Encrypt version indicates the algorithm used for encrypting the `privateKey`, the `privateKey` has its own field called privateKey.